### PR TITLE
Support deployment protection rule event type

### PR DESCRIPTION
**Pull request by BoyangHuang**
This commit adds the deployment_protection_rule event type to the list of parseable event types

Fixes: #2774.

### CODE REVIEW

File: github/event.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ func (e *Event) ParsePayload() (payload interface{}, err error) {
 		payload = &DeployKeyEvent{}
 	case "DeploymentEvent":
 		payload = &DeploymentEvent{}
+	case "DeploymentProtectionRule":
```
Comment (gmlewis) [inline github/event.go:52]:
```suggestion
	case "DeploymentProtectionRuleEvent":
```

File: github/event_types.go [hunk 2]
```diff
@@ -173,6 +173,24 @@ type DeploymentEvent struct {
 	Installation *Installation `json:"installation,omitempty"`
 }
 
+// DeploymentProtectionRuleEvent represents a deployment protection rule
```
Comment (gmlewis) [inline github/event_types.go:176]:
```suggestion
// DeploymentProtectionRuleEvent represents a deployment protection rule event.
```

File: github/event_types.go [hunk 3]
```diff
@@ -173,6 +173,24 @@ type DeploymentEvent struct {
...
+// The Webhook even name is "deployment_protection_rule".
```
Comment (gmlewis) [inline github/event_types.go:177]:
```suggestion
// The Webhook event name is "deployment_protection_rule".
```

File: github/event_types.go [hunk 4]
```diff
@@ -173,6 +173,24 @@ type DeploymentEvent struct {
...
+type DeploymentProtectionRuleEvent struct {
+	Action      *string `json:"action,omitempty"`
+	Environment *string `json:"environment,omitempty"`
+
+	// The URL Github provides for a third-party to use in order to pass/fail a deployment gate
+	DeploymentCallbackURL *string       `json:"deployment_callback_url,omitempty"`
+	Deployment            *Deployment   `json:"deployment,omitempty"`
+	Repo                  *Repository   `json:"repository,omitempty"`
+	Organization          *Organization `json:"organization,omitempty"`
+	PullRequest           *PullRequest  `json:"pull_request,omitempty"`
```
Comment (gmlewis) [inline github/event_types.go:189]:
```suggestion
	PullRequests           []*PullRequest  `json:"pull_requests,omitempty"`
```

File: github/event_types.go [hunk 5]
```diff
@@ -173,6 +173,24 @@ type DeploymentEvent struct {
...
 }
 
+// DeploymentProtectionRuleEvent represents a deployment protection rule
+// The Webhook even name is "deployment_protection_rule".
+//
+// GitHub API docs: https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#deployment_protection_rule
+type DeploymentProtectionRuleEvent struct {
+	Action      *string `json:"action,omitempty"`
+	Environment *string `json:"environment,omitempty"`
+
```
Comment (gmlewis) [inline github/event_types.go:183]:
```suggestion
Event *string `json:"event,omitempty"`
```

File: github/messages.go [hunk 6]
```diff
@@ -55,6 +55,7 @@ var (
 		"deploy_key":                     "DeployKeyEvent",
 		"deployment":                     "DeploymentEvent",
 		"deployment_status":              "DeploymentStatusEvent",
+		"deployment_protection_rule":     "DeploymentProtectionRule",
```
Comment (gmlewis) [inline github/messages.go:58]:
```suggestion
		"deployment_protection_rule":     "DeploymentProtectionRuleEvent",
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @BoyangHuang !
After a few tweaks, we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by BoyangHuang** [comment]
> Thank you, @BoyangHuang ! After a few tweaks, we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

Thank you for catching my mistakes :) much appreciated

**Comment by gmlewis** [comment]
Sorry, I forgot to mention that you will need to `gofmt` the files after you add my suggestions. Also make sure to run `go generate ./...` again then push the changes. Thank you!

**Review by gmlewis — APPROVED** [review]
Thank you, @BoyangHuang !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by BoyangHuang** [comment]
small bump here - is there any contributor available to give this a quick look? Thank you!

**Comment by gmlewis** [comment]
Hi @valbeat - do you have time for a code review?

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**